runtime/compiler/runtime/RelocationTarget.cpp

```diff
@@ -259,14 +259,30 @@ TR_RelocationTarget::patchNonVolatileFieldMemoryFence(J9ROMFieldShape* resolvedF
    }
 
 void
-TR_RelocationTarget::addPICtoPatchPtrOnClassUnload(TR_OpaqueClassBlock *classKey, void *ptr)
+TR_RelocationTarget::addPICtoPatchPtrOnClassUnload(TR_OpaqueClassBlock *classKey, void *ptr, void *metaDataBasePtr)
    {
-   platformAddPICtoPatchPtrOnClassUnload(classKey, ptr);
+   platformAddPICtoPatchPtrOnClassUnload(classKey, ptr, metaDataBasePtr);
    _reloRuntime->comp()->setHasClassUnloadAssumptions();
    }
 
 void
-TR_RelocationTarget::platformAddPICtoPatchPtrOnClassUnload(TR_OpaqueClassBlock *classKey, void *ptr)
+TR_RelocationTarget::platformAddPICtoPatchPtrOnClassUnload(TR_OpaqueClassBlock *classKey, void *ptr, void *metaDataBasePtr)
```

Comment:
I wonder if this is executed at the server. Looking at the code, this is called from `TR_RelocationTarget::addPICtoPatchPtrOnClassUnload(TR_OpaqueClassBlock *classKey, void *ptr)` which is called from a few places like:
1) TR_RelocationRecordInlinedMethod::fixInlinedSiteInfo
2) TR_RelocationRecordSymbolFromManager::activatePointer
3) TR_RelocationRecordPointer::activatePointer
4) TR_RelocationRecordClassUnloadAssumption::applyRelocation

At least some of these, if not all, are used during an AOT load which is done at the client.
@dsouzai Can you please offer AOT expertise on this one? Thanks

Reply:
I ran some sanity functional tests. Looks like so far `2. TR_RelocationRecordPointer::activatePointer()` is called at the client during AOT load. Not sure about other three yet. As mentioned on the above, @dsouzai might know more about it.

Reply:
All four of those callsites are only invoked during an AOT load.